rules/manager.go

```diff
@@ -85,6 +85,11 @@ func DefaultEvalIterationFunc(ctx context.Context, g *Group, evalTimestamp time.
 	g.setEvaluationTime(timeSinceStart)
 	g.setLastEvaluation(start)
 	g.setLastEvalTimestamp(evalTimestamp)
+
+	if g.alertStore != nil {
+		// feature enabled.
+		g.StoreKeepFiringForState()
```

Comment:
With multiple groups, `SetAlerts` will persist everything it knows, for each rule, on each evaluation cycle.
Is that OK performance-wise?  What if there are 10,000 rules?

Reply:
Note that it will only persist state for rules that have `keep_firing_for` set. 
I tested with 1,000 rules and did not notice any significant spikes in evaluation latency (no profiling done). 
There maybe higher memory/CPU usage with 10,000 rules since the state is kept in a map. We can also enforce a max limit here.